ee/app/livechat-enterprise/server/hooks/applySimultaneousChatsRestrictions.ts

```diff
@@ -0,0 +1,19 @@
+import { callbacks } from '../../../../../app/callbacks/server';
+import { LivechatDepartment } from '../../../../../app/models/server';
+import { settings } from '../../../../../app/settings/server';
+
+callbacks.add('livechat.applySimultaneousChatRestrictions', ({ departmentId }: { departmentId?: string }) => {
+	if (departmentId) {
+		const departmentLimit = departmentId ? LivechatDepartment.findOneById(departmentId)?.maxNumberSimultaneousChat : 0;
+		const departmentFilter = departmentLimit > 0 ? { 'queueInfo.chats': { $lt: Number(departmentLimit) } } : {};
```

Comment:
I don't think we're handling the condition `departmentLimit <= 0` properly here. If there is no department level `max chats` defined, then it should check for `global max chats`. If that's present then it should return that. Currently its just returning empty object so the `global max chats` setting is not coming to picture

Reply:
```suggestion
		if (departmentLimit > 0) {
			return { $match: { 'queueInfo.chats': { $lt: Number(departmentLimit) } } };
		}
```

Reply:
Can you share an example to reproduce that on my local to verify? :D

Reply:
I couldn't find any suitable example for performing a end to end test for this, however you can see this issue via adding a log [here](https://github.com/RocketChat/Rocket.Chat/pull/22611/files#diff-a8aefd768cc8ae661a2b03ac06efa3da65acd0175b869c90640a596514576fc3R69) and checking the `currentAvailableAgents` variable. I did the following on my local

1. Created 2 department (`Department 1` and `Department 2`) with `Agent 1` and `Agent 2` in each department respectively and a `common-agent` in both the department.
2. Set global waiting queue limit to `1`
3.  Start a livechat conversation. Now you'll see `currentAvailableAgents` as [`Agent 1`, `common-agent`] which is expected. Lets say the chat was assigned to `Agent 1` now.
4. Start another livechat conversation. Now also you'll see `currentAvailableAgents` as [`Agent 1`, `common-agent`] which is not correct because `Agent 1` has already exceeded the limit.

Reply:
Ahhh, I got it! That's a good observation, let me do the changes :)